File: src/python/powx-n.py
#https://leetcode.com/problems/powx-n/

class Solution:
    def myPow(self, x: float, n: int) -> float:
	    # Base case
        if n == 0:
            return 1
        # determine whether to divide by the answer 
        if n < 0: 
            div = True
        else:
            div = False
            
        n = abs(n) 
        ans = 1 
        while n>0:    
            ans *= x
            print(ans)
            n -= 1 # decrement n by 1
        
        return 1/ans if div else ans # divide by the answer if n is negative

File: src/cpp/powx-n.cpp
class Solution {
public:
    double myPow(double x, int n) {
        double ans=1;
        int base=n;
        if (base==INT_MIN&&(x==1||x==-1))
            return 1;
        else if(base==INT_MIN)
            return 0;
        n=abs(n);
        if (x==1)
            return 1;
        while(n!=0)
        {
            if(n&1)
            {
                ans=ans*x;
            }
            x=x*x;
            n=n>>1;
        }
        if(base<0)
            return 1/ans;
        else 
            return ans;
    }
};